src/CppParser/Parser.cpp

```diff
@@ -1792,7 +1792,8 @@ DeclarationContext* Parser::GetNamespace(const clang::Decl* D,
         case Decl::Record:
         {
             auto RD = cast<RecordDecl>(Ctx);
-            DC = WalkRecord(RD);
+            auto W = WalkDeclaration(RD);
```

Comment:
This whole structure saves nothing. The point is to get less code compared to before i.e:

case 1:
case 2:
case 3:
case 4:
    DC = WalkDecl();
    break;